rero_ils/modules/oauth/reroid.py

```diff
@@ -0,0 +1,139 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2019 RERO
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""RERO ID OAuth2 configuration."""
+
+from flask import redirect, url_for
+from invenio_db import db
+from invenio_oauthclient.handlers import \
+    authorized_signup_handler as base_authorized_signup_handler
+from invenio_oauthclient.handlers import oauth_error_handler
+from invenio_oauthclient.utils import oauth_link_external_id, \
+    oauth_unlink_external_id
+
+OAUTH_URL = 'https://id.rero.ch'
+REMOTE_APP = dict(
+    title='RERO ID',
+    description='RERO Identity server.',
+    icon='fa fa-user',
+    authorized_handler='rero_ils.modules.oauth.reroid'
+                       ':authorized_signup_handler',
+    disconnect_handler='invenio_oauthclient.handlers'
+                       ':disconnect_handler',
+    signup_handler=dict(
+        info='rero_ils.modules.oauth.reroid:account_info',
+        setup='rero_ils.modules.oauth.reroid:account_setup',
+        view='invenio_oauthclient.handlers:signup_handler',
+    ),
+    params=dict(
+        request_token_params={'scope': 'user:email'},
+        base_url=OAUTH_URL,
+        request_token_url=None,
+        access_token_url='{server}/oauth/token'.format(server=OAUTH_URL),
+        access_token_method='POST',
+        authorize_url='{server}/oauth/authorize'.format(server=OAUTH_URL),
+        app_key='REROID_APP_CREDENTIALS',
+    )
+)
+
+OAUTH_TEST_URL = 'https://id.test.rero.ch'
+TEST_REMOTE_APP = dict(REMOTE_APP)
```

Comment:
For my knowledge, casting into dict is for creating a new reference?

Reply:
Yes I got the inspiration from the github contrib of invenio-oauthclient.